source/virtualBuffers/gecko_ia2.py

```diff
@@ -18,10 +18,58 @@
 from comtypes import COMError
 import aria
 import config
-from NVDAObjects.IAccessible import normalizeIA2TextFormatField
+from NVDAObjects.IAccessible import normalizeIA2TextFormatField, IA2TextTextInfo, getNVDAObjectFromPoint
 
 class Gecko_ia2_TextInfo(VirtualBufferTextInfo):
 
+	def _getPointFromOffset(self,offset):
+		formatFieldStart, formatFieldEnd = self._getUnitOffsets(self.UNIT_FORMATFIELD, offset)
+		# The format field starts at the first character.
+		for field in reversed(self._getFieldsInRange(formatFieldStart, formatFieldStart+1)):
+			if not (isinstance(field, textInfos.FieldCommand) and field.command == "formatChange"):
+				# This is no format field.
+				continue
+			attrs = field.field
+			ia2TextStartOffset = attrs.get("ia2TextStartOffset")
+			if ia2TextStartOffset is None:
+				continue
+			ia2TextStartOffset += attrs.get("strippedCharsFromStart", 0)
+			relOffset = offset - formatFieldStart + ia2TextStartOffset
+			obj = self._getNVDAObjectFromOffset(offset)
+			if not issubclass(obj.TextInfo, IA2TextTextInfo):

```

Comment:
Have you tested this within a text box? The TextInfo in that case will be `MozillaCompoundTextInfo`, so this will fail. I think you need to use IA2 text directly here. See below for further thoughts on that.